shell.rb

```diff
@@ -1,33 +1,55 @@
 require('open3')
 require('pty')
+require('pathname')
+require('timeout')
 
 class Shell
+  def self.available_versions(shell)
+    ENV['PATH'].split(':').map do |d|
```

Comment:
I fix this incredibly hard to read.
How about:

```suggestion
    ENV['PATH'].split(':')
      .map { |d| File.join(d, shell) }
      .select { |f| File.file?(f) && File.executable?(f) }
      ...
```

It's also the recommended style:
> Prefer {...} over do...end for single-line blocks. Avoid using {...} for multi-line blocks. Always use do...end for “control flow” and “method definitions” (e.g. in Rakefiles and certain DSLs). Avoid do...end when chaining.

https://shopify.github.io/ruby-style-guide/